libs/github/comment.go

```diff
@@ -0,0 +1,69 @@
+package github
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/google/go-github/v63/github"
+)
+
+var (
+	ErrCommentNotFound = errors.New("comment not found")
+)
+
+type IssueIdentifier struct {
+	Owner  string
+	Repo   string
+	Number int
+}
+
+// every trait is treated as AND
+type CommentTraits struct {
+	BodyContains *string
+	UserLogin    *string
+}
```

Comment:
Forgot to comment on this. Let's avoid using pointers like this. I know the AWS Go client does this but it is really bad practice. If we need to track three states (has value, zero value, nil value) then it is better to create a specific type. The `Null*` values in the [sql package](https://pkg.go.dev/database/sql#NullBool) are an example of this.

Defining useful zero values is a Go idiom that prevents all kinds of bugs and simplifies code. See [Go Proverbs](https://go-proverbs.github.io/), specifically the excellent talk by Rob Pike: [Make the zero value useful.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=6m25s).

Reply:
After some consideration I decided that I can just use plain strings in this case.
Because matching body content or user login against empty strings makes no sense, so we can treat empty string as zero value.